File: app/src/main/java/com/otraempresa/android/example/RequestCardSubscriptionAsyncTokenAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;

import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.Transaction;

public class RequestCardSubscriptionAsyncTokenAsyncTask extends AbstractRequestCardSubscriptionAsyncTokenAsyncTask {

    RequestCardSubscriptionAsyncTokenAsyncTask(Context context) {
        super(context);
    }

    @Override
    protected Transaction requestCardSubscriptionAsyncToken(String email, String currency, String callbackUrl, String cardNumber) throws KushkiException {
        return kushki.requestCardSubscriptionAsyncToken(email,
                currency,callbackUrl, cardNumber);
    }

}

File: app/src/main/java/com/kushkipagos/kushkidemo/services/RequestCardSubscriptionAsyncTokenAsyncTask.kt
package com.kushkipagos.kushkidemo.services

import android.content.Context
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushkidemo.contracts.AbstractRequestCardSubscriptionAsyncTokenAsyncTask
import com.kushkipagos.models.Transaction

class RequestCardSubscriptionAsyncTokenAsyncTask:
    AbstractRequestCardSubscriptionAsyncTokenAsyncTask {
    constructor(context: Context): super(context)

    @Throws(KushkiException::class)
    override fun requestCardSubscriptionAsyncToken(
        email: String,
        currency: String,
        callbackUrl: String,
        cardNumber: String
    ): Transaction {
        return _kushki.requestCardSubscriptionAsyncToken(email,
            currency,callbackUrl, cardNumber)
    }
}